pkg/controller/core/workload_controller.go

```diff
@@ -129,7 +133,31 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 	if apimeta.IsStatusConditionTrue(wl.Status.Conditions, kueue.WorkloadFinished) {
 		return ctrl.Result{}, nil
 	}
+
+	cqName, cqOk := r.queues.ClusterQueueForWorkload(&wl)
+	if cqOk {
+		// because we need to react to API cluster queue events, the list of checks from a cache can lead to race conditions
+		queueAdmissionChecks, err := r.getClusterQueueAdmissionChecks(ctx, cqName)
+		if err == nil {
+			newChecks, shouldUpdate := syncAdmissionCheckConditions(wl.Status.AdmissionChecks, queueAdmissionChecks)
+			if shouldUpdate {
+				log.V(3).Info("The workload needs admission checks updates", "clusterQueue", klog.KRef("", cqName), "queueChecks", queueAdmissionChecks)
+				wl.Status.AdmissionChecks = newChecks
+				err := r.client.Status().Update(ctx, &wl)
+				return ctrl.Result{}, client.IgnoreNotFound(err)
+			}
+		}
+	}
+
 	if workload.IsAdmitted(&wl) {
```

Comment:
This is confusing here. How can a workload be both admitted and rejected at the same time?

Reply:
In case it is admitted we might request an eviction , maybe make more sense wit `evictionTriggered`.